### PR TITLE
Don't use default iface 'ovs-system' on xen setup

### PR DESCRIPTION
**Pull request by rfan1**
https://progress.opensuse.org/issues/126647

- Verification run: https://openqa.suse.de/tests/11395040#

### CODE REVIEW

**Comment by rfan1** [comment]
@alice-suse 
May I ask for your kindly review since the code changes are for `xen` hypervisor?

File: tests/installation/bootloader_svirt.pm [hunk 1]
```diff
@@ -269,6 +269,8 @@ sub run {
         # does not work.
         if (my $bridges = $svirt->get_cmd_output("virsh iface-list --all | grep -w active | awk '{ print \$1 }' | tail -n1 | tr -d '\\n'")) {
             $ifacecfg{type} = 'bridge';
+            # Due to poo#126647, the default iface 'ovs-system' on xen platform can not work fine, so we need to use br0 instead
+            $bridges = 'br0' if $bridges eq 'ovs-system';
```
Comment (alice-suse) [inline tests/installation/bootloader_svirt.pm:273]: This would additional require the setup of br0 on the host? How would make it ensured, by salt?

Comment (rfan1) [reply tests/installation/bootloader_svirt.pm:273]: Good question, on our xen host, br0 is pre-configure there: 

```#cat ifcfg-br0
STARTMODE='auto'
BOOTPROTO='dhcp'
OVS_BRIDGE='yes'
OVS_BRIDGE_PORT_DEVICE_1='eth0'
```

**Comment by alice-suse** [comment]
@nanzhg  Nan is mostly involved in svirt backend for VT test. Would you like to give it a review ?

File: tests/installation/bootloader_svirt.pm (hunk 1 above)
Comment (nicksinger) [reply tests/installation/bootloader_svirt.pm:273]: As xen currently isn't salted properly anyway I add it to https://progress.opensuse.org/issues/128222 to remind us this is covered

**Comment by nicksinger** [comment]
Why is 'ovs-system' in "bridges" in the first place? Is this some default from virsh? Otherwise it looks fine for me - I just wonder if we layer multiple workarounds on top of each other now

**Review by dzedro — APPROVED** [review]
The worker network setup is somehow wild, I would imagine to have generic default working worker network setup, without need of this additional configuration.

**Comment by rfan1** [comment]
> Why is 'ovs-system' in "bridges" in the first place? Is this some default from virsh? Otherwise it looks fine for me - I just wonder if we layer multiple workarounds on top of each other now

I would say the `ovs-system` is the default one on xen server.   however, I don't know  the background why we used 'br0' as source bridge to build VMs.
based on QA testing purpose,  seems we have worked for a long time with `br0`

**Comment by rfan1** [comment]
> The worker network setup is somehow wild, I would imagine to have generic default working worker network setup, without need of this additional configuration.

I agree with you on this point, once we have a dedicated source bridge to carry out the tests, it is easy. :)

**Comment by nanzhg** [comment]
From the job title "sle-15-SP5-JeOS-for-kvm-and-xen-Updates-x86_64-Buildrfan0619-xen_ovs-system@svirt-xen-hvm", it's related to test ovs-system on Xen, but I can't see this test cover ovs-system iface. So what's the aim of this test?

**Comment by rfan1** [comment]
> LGTM. Still a question, from the job title "sle-15-SP5-JeOS-for-kvm-and-xen-Updates-x86_64-Buildrfan0619-xen_ovs-system@svirt-xen-hvm", it's related to test ovs-system on Xen, but I can't see this test cover ovs-system iface. So what's the aim of this test?

I cloned the failed to job to run this VR to make sure my code changes can work. :) , it was not intend to test `ovs-system`:

`openqa-clone-job  --from http://openqa.suse.de --host http://openqa.suse.de  11390499 _GROUP_ID=0 -skip-download --skip-chained-deps BUILD=rfan0619 TEST=xen_ovs-system NOLOGS=1 RETRY=0 CASEDIR=https://github.com/rfan1/os-autoinst-distri-opensuse.git#xen_ovs-system`

**Pull request merged**